src/node_file.cc

```diff
@@ -833,13 +854,13 @@ static void WriteString(const FunctionCallbackInfo<Value>& args) {
 
   if (!req->IsObject()) {
     SYNC_CALL(write, nullptr, fd, &uvbuf, 1, pos)
-    if (must_free)
+    if (ownership == FSReqWrap::MOVE)
```

Comment:
You know, it's good that you commented because I just realized this code path has been leaking memory on errors for a long time now...